File: hw/maisincho/u2/HW12DataBaseInsertAndShow/Mongodb/mongo_database.py
from pymongo import MongoClient
import json


connection_string = "mongodb+srv://isaac:isaac@cluster0.xaitfht.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0"

def insertar_laptop():
    try:
        id_laptop = input("Laptop ID: ")
        name = input("Name Laptop: ")
        processor = input("Processor: ")
        ram = int(input("Enter the RAM (GB): "))
        color = input("Enter color of the laptop: ")

        document = {
            "id": id_laptop,
            "name": name,
            "processor": processor,
            "ram": ram,
            "color": color
        }

        with MongoClient(connection_string) as client:
            db = client["dbmongo"]
            collection = db["laptop"]
            collection.insert_one(document)
            print(" Laptop inserted successfully.")

    except Exception as e:
        print(f" Error connecting or inserting to MongoDB: {e}")

def mostrar_laptops():
    try:
        with MongoClient(connection_string) as client:
            db = client["dbmongo"]
            collection = db["laptop"]
            print("\n Laptops in the database:")
            for doc in collection.find():
                print(json.dumps(doc, indent=4, default=str))

    except Exception as e:
        print(f" Error retrieving data from MongoDB: {e}")

def menu():
    while True:
        print("\n  MENU:")
        print("1. Insert Laptop")
        print("2. Show Laptops")
        print("0. Exit")

        opcion = input("Select an option: ")

        if opcion == '1':
            insertar_laptop()
        elif opcion == '2':
            mostrar_laptops()
        elif opcion == '0':
            print(" Exiting program...")
            break
        else:
            print("  Invalid option, try again.")

if __name__ == "__main__":
    menu()

File: hw/maisincho/u2/HW12DataBaseInsertAndShow/DataBase/src/main/java/com/mycompany/database/MongoDataBase.java

package com.mycompany.database;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;
import java.util.Scanner;


/**
 *
 * @author Isaac Maisincho Crafters_Market DCCO ESPE
 */
public class MongoDataBase {
    static String connectionString = "mongodb+srv://isaac:isaac@cluster0.xaitfht.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0";
    
    public static void insertarLaptop() {
         Scanner scanner = new Scanner(System.in);

        System.out.print("laptop id: ");
        String id = scanner.nextLine();

        System.out.print("Name Laptop: ");
        String name = scanner.nextLine();
        
        System.out.println("Processor");
        String processor=scanner.nextLine();
        
        System.out.print("enter the RAM ");
        int ram = scanner.nextInt();
        scanner.nextLine(); 

        System.out.print("enter color laptop: ");
        String color = scanner.nextLine();

        try (MongoClient mongoClient = MongoClients.create(connectionString)) {
            MongoDatabase database = mongoClient.getDatabase("dbmongo");
            MongoCollection<Document> collection = database.getCollection("laptop");

            Document document = new Document("id", id)
                    .append("name", name)
                    .append("processor", processor)
                    .append("Ram", ram)
                    .append("color", color);

           
            collection.insertOne(document);

            System.out.println("correct insertion.");
        } catch (Exception e) {
            System.err.println("Error conection with MongoDB: " + e.getMessage());
        }
    }

    
    public static void mostrarLaptop() {
        try (MongoClient mongoClient = MongoClients.create(connectionString)) {
            MongoDatabase database = mongoClient.getDatabase("dbmongo");
            MongoCollection<Document> collection = database.getCollection("laptop");

            System.out.println("\nSaved in the database:");

          
            for (Document doc : collection.find()) {
                System.out.println(doc.toJson());
            }
        } catch (Exception e) {
            System.err.println("Error conection with MongoDB: " + e.getMessage());
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
      
        int opcion;
        do {
            
            System.out.println("\n Menu:");
            System.out.println("1. Insert Laptop ");
            System.out.println("2. Show Laptop");
            System.out.println("0. Exit");
            System.out.print("Select an option: ");
            opcion = scanner.nextInt();

           
            scanner.nextLine();

            
            switch (opcion) {
                case 1:
                    insertarLaptop();
                    break;
                case 2:
                    mostrarLaptop();
                    break;
                case 0:
                    System.out.println("Exit...");
                    break;
                default:
                    System.out.println("Invalid option, please enter a valid option..");
            }
        } while (opcion != 0);


        scanner.close();
    }
}